server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/testing/MockSmtpServerExtension.java

```diff
@@ -43,7 +44,8 @@
 
         DockerMockSmtp() {
             mockSmtpServer = DockerContainer.fromName(Images.MOCK_SMTP_SERVER)
-                .withLogConsumer(outputFrame -> LOGGER.debug("MockSMTP: " + outputFrame.getUtf8String()));
+                .withLogConsumer(outputFrame -> LOGGER.debug("MockSMTP: " + outputFrame.getUtf8String()))
+                .waitingFor(Wait.forLogMessage(".*Mock SMTP server started.*", 1));
```

Comment:
hmm as far as I can tell, forListeningPorts waits for all exposed ports to answer I'm not sure why waiting for the log is better ...

Reply:
HostPortWaitStrategy -> `Waits until a socket connection can be established on a port exposed or mapped by the container.`

I understand `any` not `all`.

A look at InternalCommandPortListeningCheck however tends to confirm what you say.

I will switch back to HostPortWaitStrategy and propose a PR upstream for fixing the ambiguous javadoc.

Reply:
(yet fixing the container logs is still valuable ;-) )

Reply:
Correction, I got misleaded by uncompiled code, testcontainer uses a single port for live check (`GenericContainer::getLivenessCheckPort`) I gonna revert the fixup and switch back to the log check.

Reply:
that's weird: https://github.com/testcontainers/testcontainers-java/blob/17b4f6c136f6f2c7dc223bad407221f62a8f0088/core/src/main/java/org/testcontainers/containers/wait/strategy/HostPortWaitStrategy.java#L26 seems to use `getLivenessCheckPorts` (with an `s`) for the external checks and 
https://github.com/testcontainers/testcontainers-java/blob/17b4f6c136f6f2c7dc223bad407221f62a8f0088/core/src/main/java/org/testcontainers/containers/wait/strategy/HostPortWaitStrategy.java#L35 uses `getExposedPorts` for the internal checks and before suggesting this I checked that this was already the case in 1.15.1 
I don't see how you could use the container if both ports are not exposed since the tests call both the smtp and the http service ...

Reply:
My fault I did clone and looked at my long outdated fork instead of recent code. Sorry for the mess.